test/libsolidity/semanticTests/functionCall/precompile_extcodesize_check.sol

```diff
@@ -0,0 +1,15 @@
+interface Identity {
+    function selectorAndAppendValue(uint value) external pure returns (uint);
+}
+contract C {
+    Identity constant i = Identity(address(0x0004));
+    function test() external pure returns (bool) {
+        // Should fail because `extcodesize(4) = 0`
+        i.selectorAndAppendValue(5);
```

Comment:
Just curious: How do precompiles work? Are they implemented in the client (host) which is why extcodesize is zero?

Reply:
> Are they implemented in the client (host)

Yes.

> which is why extcodesize is zero?

Not sure why they decided to go with zero. @axic Maybe you can tell us why they went with it?

Reply:
If you say extcodesize is nonzero, it implies that there is some code, but they do not have code that is accessible via extcodecopy.

Reply:
It is actually 0 as far as I can tell.

The reliable way to tell if something will execute or not is extcodehash: https://eips.ethereum.org/EIPS/eip-1052

Reply:
Oh that is actually a good insight, thanks!

Reply:
But wait, don't we also want to exclude existing accounts with zero code?